app/controllers/publish/courses/engineers_teach_physics_controller.rb

```diff
@@ -10,6 +10,10 @@ def new
         authorize(@provider, :can_create_course?)
         return if has_physics_subject?
 
+        if params[:goto_confirmation] && params[:course][:subjects_ids]&.include?(modern_languages_id)
```

Comment:
Might be worth putting some of these checks into smaller methods, some of the actions in this controller are getting quite large. Eg I noticed `params[:course][:subjects_ids].include?(modern_languages_id)` is also used somewhere else in this file

Reply:
Done